xonsh/proc.py

```diff
@@ -1203,12 +1203,18 @@ def parse_proxy_return(r, stdout, stderr):
     elif isinstance(r, cabc.Sequence):
         rlen = len(r)
         if rlen > 0 and r[0] is not None:
-            stdout.write(r[0])
+            if isinstance(r[0], str):
+                stdout.write(r[0])
+            else:
+                stdout.write(str(r[0]))
```

Comment:
I think this can just be `stdout.write(str(r[0]))`, because `str()` of a str object is is basically free. Same below

Reply:
I always prefer explicit over implicit, so yes, this is intentional. But OK, I can change it.

Reply:
Done.